app/controllers/registration_controller.rb

```diff
@@ -58,12 +61,16 @@ def edit
   def submit
     return if params[:iss] == ''
 
+    tenant = RailsLti2Provider::Tenant.where(uid: params[:tenant]).first
```

Comment:
@Billzzhang You are hitting the database twice for the same objects. We need to simplify it in order to improve performance.